### ISSUE TITLE
[ruby 3.0] Error: undefined method `encode' for URI:Module

### ISSUE THREAD

**Issue by erikw** [aarongustafson/jekyll-webmention_io#151]
In ruby 3.0, the `URI.encode()` seems to be removed. See a similar issue over at the module GitHub about `URI.escape()`. https://github.com/ruby/uri/issues/14

Using
- ruby: `v2.7`
- Jekyll: `v3.4`
- jekyll-webmention_io: `v3.3.5`

and compiling I already get a warning

```console
$ bundle exec jekyll build
[...]
webmention_io.rb:366: warning: URI.escape is obsolete
```

If I switch from ruby `2.7` to `3.0.1` with RVM, I get
```console
$ bundle exec jekyll build
[...]
webmention_io.rb:366:in `uri_ok?': undefined method `encode' for URI:Module (NoMethodError)
```

The mentioned line from the source is at https://github.com/aarongustafson/jekyll-webmention_io/blob/main/lib/jekyll/webmention_io.rb#L366
```ruby
uri = URI.parse(URI.encode(uri.to_s))
```

The [jekyll-webmention_io.gemspec](https://github.com/aarongustafson/jekyll-webmention_io/blob/main/jekyll-webmention_io.gemspec#L35) says 
```
s.required_ruby_version = ">= 2.3.0"
```
so I think this line must be changed to something like  `[">= 2.3.0", "< 3.0"]` until the above problem is fixed

I set up a minimal Jekyll project to demonstrate the problem. It can be cloned at https://github.com/erikw/jekyll-3.x.x-test/tree/webmentions (branch=webmentions).

**Comment by aarongustafson** [comment]
I kept having Ruby/Jekyll issues when upgrading my Mac, so I have moved off of Jekyll and will not be working on this project anymore, going forward. I am going to flag this as won’t fix, but leave it open in case someone else wants to pick up the project from here.

**Comment by fancypantalons** [comment]
The core issue, here, has been resolved with PR #150.  For that reason I'll be closing this specific issue as resolved.  That said, I've not looked at the broader issue of supporting Ruby 3.
